vcd/resource_vcd_edgegateway.go

```diff
@@ -349,6 +349,18 @@ func resourceVcdEdgeGatewayCreate(d *schema.ResourceData, meta interface{}) erro
 		},
 	}
 
+	// vCD 9.0 does not support FIPS Mode and fails if XML tag <FipsModeEnabled> is sent therefore
+	// field value must be sent only if user specified its value
+	if fipsModeEnabled, ok := d.GetOkExists("fips_mode_enabled"); ok {
+		if vcdClient.APIVCDMaxVersionIs("<= 29.0") { // vCD 9.0 or less
+			_, _ = fmt.Fprint(getTerraformStdout(), "WARNING! FIPS mode is only supported starting"+
```

Comment:
what happens here if I set? - API error? maybe is worth it here send error and fail early?

Reply:
I had a doubt about it. Here is how it looks - https://github.com/terraform-providers/terraform-provider-vcd/issues/323#issuecomment-562136931


Probably yes - I could fail early.

Reply:
If you talking about `Invalid content was found starting with element 'FipsModeEnabled'. No child element is expected at this point.` Then my answer is that 100% we have fail with good explanation. @lvirbalas and @dataclouder can include they opinions too if it is needed

Reply:
Changed